interfaces/cython/cantera/composite.py

```diff
@@ -681,8 +694,11 @@ def append(self, state=None, **kwargs):
                 raise KeyError(
                     "'{}' does not specify a full thermodynamic state".format(attr)
                 )
-            setattr(self._phase, attr, value)
-
+            if not normalize:
+                setattr(self._phase, attr[-1] + "_no_norm", value[-1])
+                setattr(self._phase, attr[:-1], value[:-1])
+            else:
+                setattr(self._phase, attr, value)
```

Comment:
The logic here would be a little easier to understand if it was reversed, in my opinion. That's for 2 reasons: 1) `normalize == True` is the default case, so it should come first and 2) dealing with negation is always trickier in the conditionals. So, it could be:

```python
if normalize:
    ...
else:
    ...
```

In addition, and related to my other comments about `*_no_norm`, I think the non-normalized case is a little bit confusing because it is using string indexing and concatenation of the `attr`. So I wonder if something like:

```python
if normalize:
    ...
else:
    if attr[-1] == "X":
        setattr(self._phase, "no norm mole fractions", value[-1])
    elif attr[-1] == "Y":
        setattr(...)
    setattr(self._phase, attr[:2], value[:2])
```
To me, this is a little bit clearer because it explicitly lists the permitted values for the last character, and also explicitly says to take the first 2 characters to set the rest of the state. It makes it easier for the next developer to come along and see what is being done here and is maybe marginally slower to execute.